### PR TITLE
Update readme

### LINKED ISSUE #33: Add in the README the LSP client which consumes the lemminx-liberty
I tested your extension with vscode-xml by using https://github.com/redhat-developer/vscode-xml/blob/master/docs/Preferences.md#extension-jars which is working well even for validation).

But I wanted to test the extension with a vscode extension which register the lemminx-liberty with package.json, but I have not found it.

Could you add it in the README please.

If you use a vscode extension which register the lemminx-liberty, I think you should be able to debug with breakpoint the lemminx-liberty extension.

### PR DESCRIPTION
**Pull request by evie-lau**
- Update readme with relevant user-information. 
- Move dev/contributing instructions into `DEVELOPING.md` and `CONTRIBUTING.md`
- Modify completion to use `setDocumentation` for markdown support, and be consistent with the hover dialog appearance
    - old, using `setDetails` 
image
    - new, using `setDocumentation` 
image
- Fixes #33

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Contributing to Liberty Config Language Server
```
Comment (cherylking) [inline CONTRIBUTING.md:1]: I don't know if this info fits what is usually in the CONTRIBUTING document. It's useful info for someone contributing, but it should be more about what the process is to [contribute](https://github.com/OpenLiberty/ci.maven/blob/main/CONTRIBUTING.md). Maybe call this document BUILDING or DEVELOPING?

File: README.md [hunk 2]
```diff
@@ -1,34 +1,44 @@
...
+and is available for use with the following clients.
+
+# Client IDEs
+* [VS Code](https://github.com/OpenLiberty/liberty-tools-vscode/tree/liberty-ls-prototype)
+* [Eclipse](https://github.com/OpenLiberty/liberty-tools-eclipse)
+* [Intellij](https://github.com/OpenLiberty/liberty-tools-intellij)
+
+## Features
+
+### Completion for Liberty server configuration files
```
Comment (cherylking) [inline README.md:15]: We need to document what is recognized as a Liberty server configuration file. Borrow from the UFO.

File: README.md [hunk 3]
```diff
@@ -1,34 +1,44 @@
...
+* Completion for Liberty properties and values 
+![Completion for Liberty properties](./docs/images/property-completion.png)
```
Comment (cherylking) [inline README.md:17]: There needs to be sufficient words that accompany each image to explain what is in the image (accessibility reasons - visually impaired). Also, I think having dark mode images may not be standard and might not fit our guidelines (again for accessibility reasons). Check with David Mueller.

**Review by cherylking — CHANGES_REQUESTED** [review]
The overall content is good. Just had a few comments.

File: README.md (hunk 2 above)
Comment (evie-lau) [reply README.md:15]: I can add some details to the main description of this repo, to include what types of Liberty config files are supported

File: README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Liberty Config Language Server
 
-Provides language server features for Liberty server configuration files.
+Provides language server features for Liberty server configuration files, which include `server.env`, `bootstrap.properties`, and `server.xml` files.
```
Comment (cherylking) [inline README.md:3]: It also includes any xml file referenced by an `include` element, and any xml file located in certain directories (confirm with UFO - but has to do with configDropins folder and the `usr/shared` folder).

File: README.md [hunk 5]
```diff
@@ -14,26 +14,34 @@ and is available for use with the following clients.
 
 ### Completion for Liberty server configuration files
 * Completion for Liberty properties and values 
-![Completion for Liberty properties](./docs/images/property-completion.png) 
-![Completion for Liberty property values](./docs/images/property-value-completion.png)
+
+![Completion feature for Liberty properties in bootstrap.properties](./docs/images/property-completion.png)
```
Comment (cherylking) [inline README.md:18]: Perhaps add a sentence other than the label such as:
When editing a bootstrap.properties file and entering a property name, completion support displays a list of potential property names that match what has been entered.

That was off the top of my head. Feel free to reword. For values, you can also mention how if a default value is defined, it is at the top of the list and is preselected.

**Review by cherylking — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (cherylking) [reply README.md:18]: So for each image, there should be a sentence or two that actually describes what the user would see.

File: README.md [hunk 6]
```diff
@@ -15,36 +15,36 @@ and is available for use with the following clients.
 ### Completion for Liberty server configuration files
 * Completion for Liberty properties and values 
 
-![Completion feature for Liberty properties in bootstrap.properties](./docs/images/property-completion.png) 
-![Completion feature for Liberty property values in bootstrap.properties](./docs/images/property-value-completion.png)
+![Screenshot of Liberty property name suggestions in a bootstrap.properties file](./docs/images/property-completion.png)
```
Comment (cherylking) [inline README.md:18]: Should we follow this pattern to have a concise title for most users of the document and a detailed description in the alt text for those users that need it?

```
![Alt text here](/path/to/img.jpg “image title”)
```

Comment (evie-lau) [reply README.md:18]: I think that's a good idea

File: CONTRIBUTING.md (hunk 1 above)
Comment (evie-lau) [reply CONTRIBUTING.md:1]: Will we need things like the Contributor License Agreement in the contribute page you linked?

File: README.md (hunk 3 above)
Comment (evie-lau) [reply README.md]: Confirmed with David that dark mode is fine, as long as it meets color contrast standards, and there are no issues there with the current screenshots.

File: CONTRIBUTING.md (hunk 1 above)
Comment (cherylking) [reply CONTRIBUTING.md:1]: > Will we need things like the Contributor License Agreement in the contribute page you linked?

Check with YK, but I think the answer will be yes.

File: README.md (hunk 4 above)
Comment (evie-lau) [reply README.md:3]: How explicit do we want to be in documenting the supported XML files?
image

I feel like this is a lot of detail for an introduction to the language server.

Comment (cherylking) [reply README.md:3]: It has to be documented somewhere for the users. Each of the IDEs will link to this for their documentation. So I think it is appropriate to fully document here what is supported. Leave out the bit about the explicit schema though. We don't actually look for that.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+Please raise any bug reports on the [Github issues page](https://github.com/OpenLiberty/liberty-language-server/issues). Be sure to search the existing issues to see if your issue has already been raised.
+
+A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong. Provide as much context as possible so we can try to recreate the issue.
+
+## Contributor License Agreement
+If you are contributing code changes via a pull request for anything except trivial changes, you must signoff on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/release/cla/open-liberty-cla-individual.pdf) If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/release/cla/open-liberty-cla-corporate.pdf). Instructions how to sign and submit these agreements are located at the top of each document. Trivial changes such as Typos, redundant spaces, minor formatting and spelling errors will be labeled as "CLA trivial", and don't require a signed CLA for consideration.
```
Comment (kathrynkodama) [inline CONTRIBUTING.md]: typos should be lowercase

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+After we obtain the signed CLA, you are welcome to open a pull request, and the team will be notified for review. We ask you follow these steps through the submission process.
+1. Ensure you run a passing local maven build explained in the [DEVELOPING](./DEVELOPING.md) before opening a PR.
+2. Open PR's against the 'main' branch.
```
Comment (kathrynkodama) [inline CONTRIBUTING.md]: Remove apostrophe "PRs"

File: DEVELOPING.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Development setup for Liberty Config Language Server
+
+This repo contains a couple projects providing IDE / language support for Open Liberty using the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/).
+
+## Projects
+
+* [lemminx-liberty](./lemminx-liberty) - an extension to the [Eclipse LemMinX](https://github.com/eclipse/lemminx) XML language server providing language features for the Liberty server.xml file.
+    * `mvn clean install` to build. Produces the `/lemminx-liberty/target/lemminx-liberty-1.0-SNAPSHOT.jar`.
+* [liberty-ls](./liberty-ls) - a language server providing language features for the Liberty bootstrap.properties and server.env files.
+    * `mvn clean install` to build. Produces the `/liberty-ls/target/liberty.ls-1.0-SNAPSHOT.jar`.
```
Comment (kathrynkodama) [inline DEVELOPING.md:10]: Could specify `liberty.ls-x.x.jar` if you don't want to update this with each version change (just a suggestion).

File: DEVELOPING.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+To test the changes interactively, you must use a language client. 
+
+Below, we will document how to build and test using the VS Code language client for Liberty Config Language Server.
+
+## Project setup in VS Code
+
+Clone the [Liberty Tools for VS Code](https://github.com/OpenLiberty/liberty-tools-vscode) repo as a sibling folder to this repo. Create a VS Code workspace these two repos at the root of the workspace. The folder structure should look something like this:
```
Comment (kathrynkodama) [inline DEVELOPING.md:18]: Missing "with": "Create a VS Code workspace with these two repos..."

File: README.md [hunk 11]
```diff
@@ -1,34 +1,56 @@
...
+![Screenshot of feature documentation appearing when hovering over a Liberty feature in a server.xml file](./docs/images/feature-hover.png "Hover on Liberty features in server.xml")
+
+### Diagnostics on Liberty server configuration files
+* Diagnostics on Liberty properties and variables
+
+![Screenshot showing diagnostics marking an invalid value for a Liberty property in a bootstrap.properties file. Hovering over the diagnostic will provide more details.](./docs/images/property-diagnostic.png "Diagnostics on Liberty properties in bootstrap.properties")
+![Screenshot showing diagnostics marking an invalid value for a Liberty variable in a server.env file. Hovering over the diagnostic will provide more details.](./docs/images/variable-diagnostic.png "Diagnostics on Liberty variables in server.env")
+* Diagnostics for Liberty features
+
+![Screenshot showing diagnostics marking an invalid feature defined in a server.xml file. Hovering over the diagnostic will provide more details.](./docs/images/feature-diagnostic.png "Diagnostics on Liberty features in server.xml")
```
Comment (kathrynkodama) [inline README.md:48]: Should we add a screenshot for diagnostics on server.xml elements that are not Liberty features?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,34 +1,23 @@
...
+After we obtain the signed CLA, you are welcome to open a pull request, and the team will be notified for review. We ask you follow these steps through the submission process.
+1. Ensure you run a passing local maven build explained in the [DEVELOPING](./DEVELOPING.md) before opening a PR.
+2. Open PR's against the `main` branch.
+3. A label will be added `CLA signed` or `CLA trivial` depending on the nature of the change.
+4. A team of "reviewers" will be notified, will perform a review, and if approved will merge the PR.
+5. If a reviewer has comments, questions or suggestions, those must be addressed before the PR is approved.
+
+## Coding Standards
+Please ensure you follow the coding standards used throughout the existing code base. Some basic rules include:
+* All files must have a Copyright including the Apache license in the header.
```
Comment (cherylking) [inline CONTRIBUTING.md:20]: Aren't we using EPL v2 for the license?

File: README.md (hunk 11 above)
Comment (evie-lau) [reply README.md:48]: I think that would be a good idea. I'm not familiar with the LemMinX side; what other diagnostic examples should we showcase?

**Comment by cherylking** [comment]
@evie-lau I just looked at the README for LT for Eclipse...perhaps our [CONTRIBUTING](https://github.com/OpenLiberty/liberty-tools-eclipse/blob/main/CONTRIBUTING.md) doc should be identical to theirs?

File: README.md (hunk 11 above)
Comment (cherylking) [reply README.md:48]: I don't think it is necessary. We've covered that diagnostics are provided for server.xml, which is the point.

**Review by cherylking — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -1,34 +1,58 @@
...
+![Screenshot showing diagnostics marking an invalid value for a Liberty variable in a server.env file. Hovering over the diagnostic will provide more details.](./docs/images/variable-diagnostic.png "Diagnostics on Liberty variables in server.env")
+* Diagnostics for Liberty XML configs
+
+![Screenshot showing diagnostics marking an invalid feature defined in a server.xml file. Hovering over the diagnostic will provide more details.](./docs/images/feature-diagnostic.png "Diagnostics on Liberty features in server.xml")
+
+### Liberty dev mode schema generation
+If [Liberty Maven Plugin](https://github.com/OpenLiberty/ci.maven) or [Liberty Gradle Plugin](https://github.com/OpenLiberty/ci.gradle) is configured with the Liberty project, Liberty Config Language Server will automatically generate a schema file based on the Liberty runtime and version to provide relevant information about the supported `server.xml` elements and Liberty features.
+
+> ### Note for dev mode for containers (devc)
```
Comment (cherylking) [inline README.md:53]: You can't refer to it as `devc` as that only applies to LMP. Need to spell out "dev mode for containers".

**Review by cherylking — APPROVED** [review]

**Pull request merged**